nodeadm/doc/api-concepts.md

```diff
@@ -0,0 +1,23 @@
+# API Concepts
+
+## Versioning
+
+The API types for `nodeadm` (the `node.eks.aws` API group) are versioned in a similar manner to the [Kubernetes API](https://kubernetes.io/docs/reference/using-api/#api-versioning).
+
+There are three levels of stability and support:
+
+### Alpha
+- Example: `v1alpha2.`
+- The software may contain bugs.
+- Support for an alpha API may be removed at any time.
+- Subsequent alpha API versions may include incompatible changes, and migration instructions may not be provided.
+
+### Beta
+- Example: `v3beta4`.
+- The software is well-tested, and production use is considered safe.
+- Support for a beta API will remain for at least one release following its deprecation.
+- Subsequent beta or stable API versions may include incompatible changes, and migration instructions will be provided.
+
+### Stable
+- Example: `v5`
+- Support for a stable API will remain at least until the release of the next major version of Amazon Linux.
```

Comment:
sounds like a reasonable promise, could there ever be a reason to break it?

Reply:
I think I can clarify this, the idea is that a stable version of the API will be supported for the life of the AL major version at its time of introduction. E.g. `v1` will be supported for AL2023, but there's no guarantee it will be supported for AL2025. When AL2025 is released, we are permitted to require `v2` for AL2025 nodes.

Reply:
ah ok gotcha